snuba/query/snql/parser.py

```diff
@@ -683,6 +685,12 @@ def visit_select_list(
         ret.append(right_column)
         return ret
 
+    def visit_arrayjoin_clause(
+        self, node: Node, visited_children: Tuple[Any, Any, Any, Expression]
+    ) -> Expression:
+        _, _, _, expression = visited_children
```

Comment:
This expression cannot have an alias under any circumstance, right ?

Reply:
Not according the to the regex. The only expression that supports an alias is a function and those are not allowed in the array join (currently).